src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BookingSearchService.kt

```diff
@@ -44,28 +44,40 @@ class BookingSearchService(
       probationRegionId,
       buildPage(sortOrder, sortField, page),
     )
-    var results = updatePersonNameFromOffenderDetail(findBookings, user)
+    var results = removeRestrictedAndUpdatePersonNameFromOffenderDetail(
+      mapToBookingSearchResults(findBookings),
+      user,
+    )
     if (sortField == BookingSearchSortField.personName) {
       results = sortBookingResultByPersonName(results, sortOrder)
     }
 
     return Pair(results, getMetadata(findBookings, page))
   }
 
-  private fun updatePersonNameFromOffenderDetail(
-    findBookings: Page<BookingSearchResult>,
+  private fun removeRestrictedAndUpdatePersonNameFromOffenderDetail(
+    bookingSearchResultDtos: List<BookingSearchResultDto>,
     user: UserEntity,
-  ) = mapToBookingSearchResults(findBookings)
-    .mapNotNull { result ->
-      val offenderDetails =
-        when (val offenderDetailsResult = offenderService.getOffenderByCrn(result.personCrn, user.deliusUsername)) {
-          is AuthorisableActionResult.Success -> offenderDetailsResult.entity
-          is AuthorisableActionResult.Unauthorised -> return@mapNotNull null
-          is AuthorisableActionResult.NotFound -> null
+  ): List<BookingSearchResultDto> {
+    val offenderSummaries = offenderService.getOffenderSummariesByCrns(
+      bookingSearchResultDtos.map { it.personCrn },
+      user.deliusUsername,
+      ignoreLao = false,
+      forceApDeliusContextApi = false,
+    )
+
+    return bookingSearchResultDtos
+      .map { result -> result to offenderSummaries.first { it.crn == result.personCrn } }
+      .filter { (_, offenderSummary) -> offenderSummary !is PersonSummaryInfoResult.Success.Restricted }
+      .map { (result, offenderSummary) ->
+        result
+        result.personName = when (offenderSummary) {
+          is PersonSummaryInfoResult.Success.Full -> "${offenderSummary.summary.name.forename} ${offenderSummary.summary.name.surname}"
```

Comment:
we _could_ use the new OffenderDetailsUtil#getNameFromPersonSummaryInfoResult here which would be cleaner, but It would slightly change the behaviour:

* we'd include middle names in the result (presumably this is a good thing)
* for unknown we'd return "Unknown" (instead of null)
* for not found we'd return "Not Found" (instead of null)

I tried to look in the corresponding UI code and couldn't figure out if nulls in the later two values are relied upon to control logic in the UI (e.g. hide a row?).

Any thoughts from a reviewer?